xbmc/cores/VideoPlayer/MediaPipelineWebOS.cpp

```diff
@@ -345,8 +351,12 @@ bool CMediaPipelineWebOS::OpenVideoStream(CDVDStreamInfo hint)
     {
       std::scoped_lock lock(m_videoCriticalSection);
       SetupBitstreamConverter(hint);
+      if (m_videoHint.uniqueId != hint.uniqueId ||
+          (m_videoHint.width == hint.width && m_videoHint.height == hint.height))
+      {
+        Flush(true);
+      }
```

Comment:
I am failing to understand this condition. Adaptive streaming means that we can change bitrate, resolution and framerate dynamically. So why this check for the same width and the uniqueId?
Instead, can we remove flushing here completely then change the condition above like this:

```c++
if (m_videoHint.codec == hint.codec && m_videoHint.hdrType == hint.hdrType)
```

Reply:
Still, I think the condition can be more relaxed as in my comment above

Reply:
What I meant is completely remove the flushing from this path. Why do we care if it's a different stream. Adaptively changing the bitrate by switching to another stream for instance does not require a flush.